### DISCUSSION TITLE
Retries in AsyncClient is not working for me

### DISCUSSION THREAD

**Discussion by mhihasan** [encode/httpx#1745] [Potential Issue]
I am trying to access an API which alternates 200 and 500 status. That's why I need to retry once in case it returns 500 error, but for 200 response I don't need to retry.

I am trying the following way, but getting `<Response [500 Server Error]>` and `<Response [200 Ok]>` alternatively.
I need `<Response [200 Ok]>` always.
```python
import asyncio

import httpx


async def get_device(device_id: str):
    transport = httpx.AsyncHTTPTransport(retries=1)
    async with httpx.AsyncClient(transport=transport) as client:
        response = await client.get(f"http://example.com/device_id/{device_id}")
    print(response) # Output: alternating `<Response [500 Server Error]>` and `<Response [200 Ok]>`


if __name__ == '__main__':
    asyncio.run(get_device("xcvbrty123"))

```

**Comment by lovelydinosaur** [comment]
> I am trying to achieve this with the following way, but it is not working for me

You'd need to describe *what's* not working for someone to be able to help you out with this.

Probably start by telling us exactly *which* kind of exception is being raised?

**Reply by mhihasan** [reply]
I have added more desccription on the problem.

**Reply by lovelydinosaur** [reply]
Right, so `retries=...` [gives you *connection retries*])(https://www.python-httpx.org/advanced/#custom-transports). Ie, if an HTTP connection couldn't be established to the server then it'll retry that.

We don't have any support for automatically retiring requests based on `5xx` status codes. There's a similar discussion here... https://github.com/encode/httpx/discussions/1755
